mono/metadata/gc.c

```diff
@@ -885,7 +885,9 @@ mono_gc_cleanup (void)
 
 				/* Set a flag which the finalizer thread can check */
 				suspend_finalizers = TRUE;
-
+#ifdef HAVE_SGEN_GC
```

Comment:
Could you add that function to both boehm and null-gc and remove this one define?

The GC API between the 2 GCs should converge and not the other way around.